### PR TITLE
Feat/add new character types

### PR DESCRIPTION
**Pull request by Dakayl**
Ok that's a big one... I will a lot of comments here and there to explain.

### CODE REVIEW

File: css/vtm5e.css [hunk 1]
```diff
@@ -540,6 +553,14 @@
...
   max-width: 10%;
+  margin: 1px;
+	background: rgba(255,255,255,0.5);
+	border-radius: 3px;
+}
+
+.roll-img.hunger-dice {
+  background: rgba(255,230,230,0.5);
+  border-color: #790813;
 }
```
Comment (Dakayl) [inline css/vtm5e.css:565]: I was looking for a more distinct look of hunger dice for the chat :)

File: css/vtm5e.css [hunk 2]
```diff
@@ -550,3 +571,20 @@
...
+  position: absolute;
+  content: 'X';
+  font-size: 24px;
+  color: red;
+  margin-left: -20px;
+  margin-top: -2px;
+}
+
+#sidebar.app {
+	background: rgba(20,0,0,0.9);
```
Comment (Dakayl) [inline css/vtm5e.css]: I thought a little red background was better :)

File: lang/fr.json [hunk 3]
```diff
@@ -77,7 +83,7 @@
     "VTM5E.AddItem": "Ajouter",
     "VTM5E.Edit": "Éditer",
     "VTM5E.Delete": "Supprimer",
-    "VTM5E.CustomRolls": "Jetes Personnalisés",
+    "VTM5E.CustomRolls": "Jets Personnalisés",
```
Comment (Dakayl) [inline lang/fr.json:86]: A little typo here. It's jets, not jetes.

File: lang/fr.json [hunk 4]
```diff
@@ -2,6 +2,12 @@
     "VTM5E.Name": "Nom",
     "VTM5E.Clan": "Clan",
     "VTM5E.Predator": "Prédateur",
+
+    "VTM5E.Vampire": "Vampire",
+    "VTM5E.Coterie": "Coterie",
+    "VTM5E.Ghoul": "Goule",
+    "VTM5E.Mortal": "Mortel",
+    "VTM5E.Domitor": "Domitor",
```
Comment (Dakayl) [inline lang/fr.json:10]: Add some usefull terminology - French official wording

File: lang/en.json [hunk 5]
```diff
@@ -2,7 +2,13 @@
     "VTM5E.Name": "Name",
     "VTM5E.Clan": "Clan",
     "VTM5E.Predator": "Predator",
-    
+
+    "VTM5E.Vampire": "Vampire",
```
Comment (Dakayl) [inline lang/en.json:6]: Add some usefull terminology

File: module/actor/actor.js [hunk 6]
```diff
@@ -34,4 +34,63 @@ export class VampireActor extends Actor {
...
+
+
+   /**
+   * Redefines the create "actor" type with translations :)
+   * @param {object} data         Initial data with which to populate the creation form
+   * @param {object} [options]    Positioning and sizing options for the resulting dialog
+   * @return {Promise<Document>}  A Promise which resolves to the created Document
+   * @memberof ClientDocumentMixin
+   */
+     static async createDialog(data={}, options={}) {
```
Comment (Dakayl) [inline module/actor/actor.js]: Why do I recreate the createDialog box ?

Two reasons ? 
1. I have replaced the "character" actor type with a more clear "vampire" type. We can't delete yet the old character type but still we should'nt create new "character".

2. Translated terms, yay !

File: module/actor/coterie-actor-sheet.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
+/* global ActorSheet, game, mergeObject, duplicate */
```
Comment (Dakayl) [inline module/actor/coterie-actor-sheet.js]: the coterie sheet has the most basic things :
Features, Notes, Equipement & Biography

File: module/actor/ghoul-actor-sheet.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
+/* global Dialog, game, mergeObject */
```
Comment (Dakayl) [inline module/actor/ghoul-actor-sheet.js:1]: Ghouls are mortal with some vampiric powers - and a domitor - so we can inherit the MortalActorSheet

File: module/actor/mortal-actor-sheet.js [hunk 9]
```diff
@@ -0,0 +1,375 @@
+/* global DEFAULT_TOKEN, Dialog, duplicate, game, mergeObject */
```
Comment (Dakayl) [inline module/actor/mortal-actor-sheet.js:1]: I did inherit mortalActorSheet from the Coterie Sheet : features, notes and equipement are still there. We just add stats and a few more things.

File: module/actor/ghoul-actor-sheet.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+    }).render(true)
+  }
+
+  
+  _onVampireRoll (event) {
+    event.preventDefault()
+    const element = event.currentTarget
+    const dataset = element.dataset
+    const item = this.actor.items.get(dataset.id)
+    let disciplineValue = 1
```
Comment (Dakayl) [inline module/actor/ghoul-actor-sheet.js]: Important stuff : Ghouls only has 1 die for disciplines, even if they can learn powers.

File: module/actor/roll-dice.js [hunk 11]
```diff
@@ -0,0 +1,110 @@
+
+/* global ChatMessage, Roll, game */
```
Comment (Dakayl) [inline module/actor/roll-dice.js:2]: The roll dice function is cool, but I thought  we should put it on its own place :)

File: module/actor/roll-dice.js [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      label = label + '<img src="systems/vtm5e/assets/images/normal-fail.png" alt="Normal Fail" class="roll-img normal-dice" />'
+    }
+  
+    label = label + '<br>'
+  
+    for (let i = 0, j = hungerCritSuccess; i < j; i++) {
+      label = label + '<img src="systems/vtm5e/assets/images/red-crit.png" alt="Hunger Crit" class="roll-img hunger-dice" />'
+    }
+    for (let i = 0, j = hungerSuccess; i < j; i++) {
+      label = label + '<img src="systems/vtm5e/assets/images/red-success.png" alt="Hunger Success" class="roll-img hunger-dice" />'
```
Comment (Dakayl) [inline module/actor/roll-dice.js]: hunger-dice class for hunger-dice design yay :)

File: module/actor/vampire-actor-sheet.js [hunk 13]
```diff
@@ -0,0 +1,193 @@
+/* global game, mergeObject */
+
+import { GhoulActorSheet } from './ghoul-actor-sheet.js'
```
Comment (Dakayl) [inline module/actor/vampire-actor-sheet.js:3]: So a Vampire is mostly a ghoul with hunger, blood potency and better disciplines :)

File: module/main.js [hunk 14]
```diff
@@ -2,21 +2,35 @@
...
+import { rollDice } from './actor/roll-dice.js'
+import { CoterieActorSheet } from './actor/coterie-actor-sheet.js';
+import { MortalActorSheet } from './actor/mortal-actor-sheet.js'
+import { GhoulActorSheet } from './actor/ghoul-actor-sheet.js'
+import { VampireActorSheet } from './actor/vampire-actor-sheet.js'
 
 Hooks.once('init', async function () {
   console.log('Initializing Schrecknet...')
 
+  game.settings.register('vtm5e', 'worldVersion', {
```
Comment (Dakayl) [inline module/main.js:19]: I'm trying to insert a migration logic system.

We set a value to current version (1.5) and when that version changes... we can update.

File: module/main.js [hunk 15]
```diff
@@ -33,8 +47,25 @@ Hooks.once('init', async function () {
 
   // Register sheet application classes
   Actors.unregisterSheet('core', ActorSheet)
+
```
Comment (Dakayl) [inline module/main.js:50]: So many sheets <3

File: module/migration.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
+/* global ui, game */
+
+let worldVersion;
+
+export const migrateWorld = async () => {
```
Comment (Dakayl) [inline module/migration.js:5]: Allow to migrate some data. We only use it once to migrate old characters to vampire type. 
But It's a useful tool to have, to change data.

File: module/templates.js [hunk 17]
```diff
@@ -7,6 +7,22 @@ export const preloadHandlebarsTemplates = async function () {
   console.log('Schrecknet : loading subroutines')
   // Define template paths to load
   const templatePaths = [
+    // Actor Sheet Partials
+    'systems/vtm5e/templates/actor/parts/biography.html',
```
Comment (Dakayl) [inline module/templates.js:11]: well... I love parts.

File: css/vtm5e.css [hunk 18]
```diff
@@ -277,6 +283,13 @@
   width: 100px;
   margin-right: 0;
 }
+.vtm5e .sheet-header .profile-img-zone h1 {
+  color: #790813;
+  font-family: "cormorantlight", sans-serif;
+  text-transform: uppercase;
+  font-size: 20px;
+  font-weight: bold;
+}
```
Comment (Dakayl) [inline css/vtm5e.css:293]: Sheet has a little title like "Coterie" or "Ghoul" to distinct each type.

File: template.json [hunk 19]
```diff
@@ -1,8 +1,27 @@
 {
   "Actor": {
-    "types": ["character"],
+    "types": ["character", "vampire", "ghoul", "mortal", "coterie"],
```
Comment (Dakayl) [inline template.json:3]: characters are still heare... for the moment.

File: templates/actor/coterie-sheet.html [hunk 20]
```diff
@@ -0,0 +1,38 @@
+<form class="{{cssClass}} flexcol {{#if locked}}locked{{/if}} coterie-sheet" autocomplete="off">
```
Comment (Dakayl) [inline templates/actor/coterie-sheet.html:1]: Coteries, Ghoul and Mortal sheets need a better design.

This is functionnal but not very stylish.

File: css/vtm5e.css [hunk 21]
```diff
@@ -563,3 +585,7 @@
   margin-left: -20px;
   margin-top: -2px;
 }
+
+#sidebar.app {
+  background: rgba(20, 0, 0, 0.9);
+}
```
Comment (Dakayl) [inline css/vtm5e.css:591]: I thought a little red was nicer :)

**Review by Dakayl — COMMENTED** [review]
Resolves https://github.com/Rayji96/foundry-V5/issues/95 

And maybe a bit of https://github.com/Rayji96/foundry-V5/issues/13. I treat haven as a basic backgrounds but we can improve that later.

File: css/vtm5e.css (hunk 1 above)
Comment (Dakayl) [reply css/vtm5e.css:565]: demo_hunger_dice

File: css/vtm5e.css (hunk 2 above)
Comment (Dakayl) [reply css/vtm5e.css]: demo_redish_panel

File: templates/actor/coterie-sheet.html (hunk 20 above)
Comment (Dakayl) [reply templates/actor/coterie-sheet.html:1]: coterie-sheet

File: css/vtm5e.css (hunk 21 above)
Comment (Dakayl) [reply css/vtm5e.css:591]: demo_redish_panel

**Comment by Rayji96** [comment]
This is going to take dedicated time to review lol. I'll try to review it all in 2 days

**Comment by Rayji96** [comment]
Merge conflict from **https://github.com/Rayji96/foundry-V5/pull/91**
If you feel like it can you refactor it a bit? I wonder if there's a way to combine blood potency dict with the number dict that was added

**Comment by NbrouetEuronews** [comment]
Take your time to review this a big one :)

Yeah i'll will resolve the conflict and refactor that part :)

**Comment by Dakayl** [comment]
Yes, this is a big one, take your time. :)

About  #91, I guess I can refactor it a bit, yeah. I was'nt a big fan of the blood potency array in the middle of a function, so either way a clean up will be nice.

**Comment by MarkusOzgur** [comment]
>About #91, I guess I can refactor it a bit, yeah. I was'nt a big fan of the blood potency array in the middle of a function, so either way a clean up will be nice.

To be perfectly honest, I hate how I ended up doing it as well, but apparently, there isn't a good way to access the already present array used for localization and the prepare function is the way to go for sheet data (at least it is what people over the discord server told me). If a cleaner way of doing it is present and I was just dumb, it would be much more ideal :D

**Comment by Dakayl** [comment]
No you're not dumb, this is kinda tricky. I will look about how we can handle it better, tonight. I'm not even sure of how.

**Comment by Dakayl** [comment]
Ok I have fixed the merge conflict I guess.
And also I've bugfixed something : willpower rolls where with 3 dices only (why?) an used hunger dices. They are supposed not to.

**Comment by Rayji96** [comment]
Willpower rolls were made as Willpower rerolls. I think we have in-chat willpower rerolls now so willpower rolls can be regular willpower rolls. When I initially made the system, I made it with my group in mind and what rolls we used the most.

**Comment by Dakayl** [comment]
So we have both, sounds good :)

**Comment by Rayji96** [comment]
Having more than one reviewer would be a blessing but I'm going to go ahead and merge. Test some things too :)

**Pull request merged**